packages/agent/src/tool/read-file/ReadFileTool.ts

```diff
@@ -36,7 +39,8 @@ export const readFile = ({
       filePath: zod.string(),
     }),
     outputSchema: zod.object({
-      content: zod.string(),
+      content: zod.string().optional(),
+      error: zod.string().optional(),
```

Comment:
This would lead to a succesful Step with an error. Throwing an error with a good message might be better, because  it would create the right result type in packages/agent/src/tool/ToolStep.ts

Reply:
Quite possible - error handling is still largely a TODO. Let's get this PR in and then I'll clean up the error handling later.